packages/gooddata-eval/src/gooddata_eval/core/agentic/metric_skill.py

```
@@ -204,9 +197,14 @@ def _execute_single_metric_run(
                 metric_id_to_delete = candidate.get("metric_id")
                 break
             response_text = (chat_result.text_response or "").strip()
-            if _is_asking_clarification(response_text):
+            if not response_text and not chat_result.tool_call_events:
+                break
+            if _iteration >= max_iterations - 1:
+                break
+            try:
                 current_question = generate_simulated_response(response_text, primary_expected)
-            else:
+            except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)